server/src/graql/executor/QueryExecutor.java

```diff
@@ -378,10 +379,12 @@ public ConceptSet delete(GraqlDelete query) {
             } else {
                 throw GraknServerException.create("Unhandled concept type isn't a schema concept or a thing");
             }
-        });
+            // TODO rewrite to either throw or return an ID
+            return null; // we shouldn't be hitting this case without throwing an exception
+        }).filter(Objects::nonNull).collect(Collectors.toSet());
```

Comment:
@marco-scoppetta @kasper-piskorski maybe you guys have an idea how to reorganise this loop to avoid the `nonNull`

Reply:
if you want to avoid the `null` thing just keep the `.forEach` instead of the `stream().map()` and instead of `return id` you can `set.add(id)`

Reply:
better!